core/src/parquet/read/column.rs

```diff
@@ -124,19 +129,81 @@ impl ColumnReader {
                             bit_width,
                             is_signed,
                         } => match (bit_width, is_signed) {
-                            (8, true) => typed_reader!(Int8ColumnReader, Int8),
+                            (8, true) => match promotion_info.physical_type {
+                                PhysicalType::FIXED_LEN_BYTE_ARRAY => {
+                                    if promotion_info.precision <= DECIMAL_MAX_INT_DIGITS
+                                        && promotion_info.scale < 1
```

Comment:
Here and elsewhere - What is this condition `promotion_info.scale < 1` for? To make sure this is an integer type? We don't check it for the `promotion_info.precision <= DECIMAL_MAX_LONG_DIGITS` case.
Also, do we want to allow `scale < 0` here, not sure if we will actually apply it correctly?

Reply:
Yes, when `promotion_info.scale < 1`, it can be treated as regular integer.
Perhaps I can do the same for  the promotion_info.precision <= DECIMAL_MAX_LONG_DIGITS case. It is an optimization.
Spark guarantees not to have `scale < 0` https://github.com/apache/spark/pull/26881

Reply:
> Spark guarantees not to have `scale < 0` [apache/spark#26881](https://github.com/apache/spark/pull/26881)

👍🏽